amy/workshops/models.py

```diff
@@ -970,6 +970,26 @@ def save(self, *args, **kwargs):
         self.twitter = self.twitter or None
         super().save(*args, **kwargs)
 
+    def archive(self) -> None:
+        """
+        Archives the Person.
+
+        When archiving all personal information associated with the user profile
+        should be deleted except for first name, last name and their teaching history.
+        """
+        self.email = None
+        self.country = ""
+        self.airport = None
+        self.github = None
+        self.twitter = None
+        self.url = ""
+        self.user_notes = ""
+        self.affiliation = ""
+        self.is_active = False
+        self.occupation = ""
+        self.orcid = ""
+        self.save()
```

Comment:
Does this remove M2M relationships like knowledge domains, tasks, or awards?

Reply:
@pbanaszkiewicz So I was told we wanted to keep things like awards and qualifications, so that in the event the user signed up again we'd have them. Though matching the user up with this information well might be a bit difficult.  hmm I should add a ticket for thinking that through.

Reply:
Gotcha, I see the purpose now.